File: sdk/php/src/Internal/Interfaces/WebSocketMessageCallback.php
<?php

namespace KuCoin\UniversalSDK\Internal\Interfaces;

use JMS\Serializer\Serializer;
use KuCoin\UniversalSDK\Model\WsMessage;

interface WebSocketMessageCallback
{

    /**
     * Handles incoming WebSocket messages.
     * @param WsMessage $message
     * @param Serializer $serializer
     * @return mixed
     */
    public function onMessage(WsMessage $message, Serializer $serializer);

}

File: sdk/java/src/main/java/com/kucoin/universal/sdk/internal/interfaces/WebSocketMessageCallback.java
package com.kucoin.universal.sdk.internal.interfaces;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.kucoin.universal.sdk.model.WsMessage;

public interface WebSocketMessageCallback {

  /** Handles incoming WebSocket messages. */
  void onMessage(WsMessage message, ObjectMapper objectMapper);
}
